cl/blockbuilder/blockbuilder.go

```diff
@@ -153,50 +165,19 @@ func (bb *BlockBuilder) GetPayload(ctx context.Context) error {
 		)
 	}
 
-	// Load execution head to get previous block timestamp
-	err = util.RetryWithBackoff(ctx, maxAttempts, bb.logger, func() error {
-		head, err = bb.loadExecutionHead(ctx)
-		if err != nil {
-			bb.logger.Warn(
-				"Failed to load execution head, retrying...",
-				"error", err,
-			)
-			return err // Will retry
-		}
-		return nil // Success
-	})
-	if err != nil {
-		return fmt.Errorf("latest execution block: %w", err)
-	}
-
-	prevTimestamp := head.BlockTime
-
-	var ts uint64
-
-	if bb.lastBlockTime.IsZero() {
-		// First block, initialize LastCallTime and set default timestamp
-		ts = uint64(time.Now().UnixMilli()) + bb.buildDelayMs
-	} else {
-		// Compute diff in milliseconds
-		diff := currentCallTime.Sub(bb.lastBlockTime)
-		diffMillis := diff.Milliseconds()
-
-		if uint64(diffMillis) <= bb.buildDelayMs {
-			ts = prevTimestamp + bb.buildDelayMs
-		} else {
-			// For every multiple of buildDelay that diff exceeds, increment the block time by that multiple.
-			multiples := (uint64(diffMillis) + bb.buildDelayMs - 1) / bb.buildDelayMs // Round up to next multiple of buildDelay
-			ts = prevTimestamp + multiples*bb.buildDelayMs
-		}
-	}
-
-	// Very low chance to happen, only after restart and time.Now is broken
-	if ts <= head.BlockTime {
-		ts = head.BlockTime + 1 // Subsequent blocks must have a higher timestamp.
+	ts := uint64(time.Now().UnixMilli())
+	if ts <= bb.executionHead.BlockTime {
+		bb.logger.Warn(`Leader: Current timestamp is less than or equal to
+			previous block timestamp. Genesis timestamp could have been set incorrectly`,
+			"current_ts", ts,
+			"prev_head_block_time", bb.executionHead.BlockTime,
+		)
+		ts = bb.executionHead.BlockTime + 1
+		time.Sleep(time.Until(time.UnixMilli(int64(ts))))
```

Comment:
Better to check for ctx cancellation than just sleeping


```suggestion
		select {
		case <-ctx.Done():
		    // Handle Ctx cancellation
		case <-time.After():
		}
```